tests/unittests/sources/azure/test_errors.py

```diff
@@ -254,4 +254,4 @@ def test_imds_invalid_metadata():
 
     assert error.reason == "invalid IMDS metadata for key=compute"
     assert error.supporting_data["key"] == key
-    assert error.supporting_data["value"] == repr(value)
+    assert error.supporting_data["value"] == value
```

Comment:
Without repr() we lose typing info.  Not really a problem given the context of the error, but we can get the best of both worlds by adding `type=type(value).__name__`  to supporting data.